### PR TITLE
PluginExtensions: Made it possible to control modal size from extension

### PR DESCRIPTION
**Pull request by mckn**
**What is this feature?**

This PR will allow plugins that use UI extensions to control the size of the modal that will be displayed when calling `openModal` when an user trigger an UI extension. 

You can specify `width` and `height` in either pixels or percent by passing it as options to the `openModal` function.

**Why do we need this feature?**

When opening more advanced flows when the user triggers an UI extension there is a need to control the size of the Modal.

**Who is this feature for?**

Plugin developers

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/extensions/utils.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
+import { css } from '@emotion/css';
 import { isArray, isObject } from 'lodash';
-import React from 'react';
+import React, { useMemo } from 'react';
```
Comment (mckn) [inline public/app/features/plugins/extensions/utils.tsx:3]:
```suggestion
import React from 'react';
```

**Review by leventebalogh — APPROVED** [review]
Nice! 👏

File: public/app/features/plugins/extensions/utils.tsx [hunk 2]
```diff
@@ -42,28 +44,38 @@ export function handleErrorsInFn(fn: Function, errorMessagePrefix = '') {
...
+const getModalWrapper = ({
   // The title of the modal (appears in the header)
   title,
   // A component that serves the body of the modal
   body: Body,
-}: Parameters<PluginExtensionEventHelpers['openModal']>[0]) => {
+  width,
+  height,
+}: PluginExtensionOpenModalOptions) => {
+  const className = css({ width, height });
```
Comment (academo) [inline public/app/features/plugins/extensions/utils.tsx:74]: what happens if these variables are undefined? (not passed as parameters)

**Review by academo — APPROVED** [review]

File: public/app/features/plugins/extensions/utils.tsx (hunk 2 above)
Comment (mckn) [reply public/app/features/plugins/extensions/utils.tsx:74]: I can add an assertion for that in one of the tests but they will be ignored. In this case they will create an empty style block.

**Review by leventebalogh — APPROVED** [review]
🚀

**Review by academo — APPROVED** [review]
nice test addition

**Pull request merged**